pysteps/blending/linear_blending.py

```diff
@@ -180,24 +184,70 @@ def forecast(
             # Set weights at times before start_blending and after end_blending
             if weight_nwp < 0.0:
                 weight_nwp = 0.0
+                if n_ens_members_max == 1:
+                    R_blended[i, :, :] = precip_nowcast[i, :, :]
+                else:
+                    R_blended[:, i, :, :] = precip_nowcast[:, i, :, :]
             elif weight_nwp > 1.0:
                 weight_nwp = 1.0
-
-            # Calculate weight_nowcast
-            weight_nowcast = 1.0 - weight_nwp
-
-            # Calculate output by combining precip_nwp and precip_nowcast,
-            # while distinguishing between ensemble and non-ensemble methods
-            if n_ens_members_max == 1:
-                R_blended[i, :, :] = (
-                    weight_nwp * precip_nwp[i, :, :]
-                    + weight_nowcast * precip_nowcast[i, :, :]
-                )
+                if n_ens_members_max == 1:
+                    R_blended[i, :, :] = precip_nwp[i, :, :]
+                else:
+                    R_blended[:, i, :, :] = precip_nwp[:, i, :, :]
             else:
-                R_blended[:, i, :, :] = (
-                    weight_nwp * precip_nwp[:, i, :, :]
-                    + weight_nowcast * precip_nowcast[:, i, :, :]
-                )
+                # Calculate weight_nowcast
+                weight_nowcast = 1.0 - weight_nwp
+
+                # Calculate output by combining precip_nwp and precip_nowcast,
+                # while distinguishing between ensemble and non-ensemble methods
+                if saliency:
+                    if n_ens_members_max == 1:
+                        R_blended[i, :, :] = (
+                            get_ws(
+                                weight_nowcast,
+                                precip_nowcast[i, :, :],
+                                precip_nwp[i, :, :],
+                            )
+                            * precip_nowcast[i, :, :]
+                            + (
+                                1
+                                - get_ws(
```

Comment:
you're computing twice the same weights, what about adding a call to `get_ws` before applying the blending? for example

```
weights_saliency = get_ws(...)
R_blended = weights_saliency * precip_nowcast + (1 - weights_saliency) * precip_nwp
```

Reply:
In the paper they use different weights to calculate ws. However, following the approach of the paper, the values of the salient weight won't add up to 1. It is now changed by only calculating ws once.